figo/figo.py

```diff
@@ -139,20 +139,26 @@ class FigoException(Exception):
     They consist of a code-like `error` and a human readable `error_description`.
     """
 
-    def __init__(self, error, error_description):
+    def __init__(self, error, error_description, code=None):
         """Create a Exception with a error code and error description."""
         message = u"%s (%s)" % (error_description, error)
         super(FigoException, self).__init__(message)
 
         # XXX(dennis.lutter): not needed internally but left here for backwards compatibility
+        self.code = code
         self.error = error
         self.error_description = error_description
 
     @classmethod
     def from_dict(cls, dictionary):
         """Helper function creating an exception instance from the dictionary returned
         by the server."""
-        return cls(dictionary['error']['message'], dictionary['error']['description'])
+        if 'code' in dictionary['error']:
+            code = dictionary['error']['code']
+        else:
+            code = None
+
+        return cls(dictionary['error']['message'], dictionary['error']['description'], code)
```

Comment:
for `code` i would use `dictionary['error'].get('code')` and remove the lines above